app/api/validations/user.py

```diff
@@ -53,6 +53,15 @@ def validate_user_registration_request_data(data):
     if not is_valid[0]:
         return is_valid[1]
 
+    # validate optional fields
+    need_mentoring = data.get('need_mentoring', None)
+    if need_mentoring is '' or (need_mentoring and not isinstance(need_mentoring, bool)):
```

Comment:
Not sure about this condition. I had a bug previously regarding something similar to this. Because in Python if you do if (boolean_value), if the value is false, that won't go to the if. So we gotta be careful between the difference of a None value and a false value. We want to accept a `need_mentoring` or `available_to_mentor` if they are set to false. Make sure you do tests that test this. Am I being confusing?

Reply:
The value won't be ' ' aka empty string, because you explicitly tell that this will be a boolean. So you don't need to check if its an empty string right? Both here and in the next if condition

Reply:
Since the fields are optional you just need to check, if `need_mentoring` is not None, then it has to be a boolean. So `if need_mentoring != None and not isinstance(need_mentoring, bool)` is the right condition? Does this make sense? If it does, the condition will be similar for the `availability_to_mentor` field

Reply:
Thank you @isabelcosta , thank you for pointing out the duplication. I realized that the decorator `@expect` accepts `validate=True` as a parameter that specifies if the request data should be validated. This ensures that the fields only accept booleans.

Reply:
Yes! I'm glad you understood it from the code :D